*: avoid constructing `AmbientContext` without a constructor function

knz opened the pull request:
Informs  #72815: if we're going to use a mandatory constructor, the first step is to ensure that code doesn't instantiate AC nilly-willy.

knz committed:
cli: attach the drain process to tracing

This fixes a mistaken use of an empty AmbientContext.

Release note: None

knz committed:
log: add a dummy ambient ctx constructor for tests

Release note: None

knz committed:
acceptance,rpc: use test ambientcontexts where appropriate

Release note: None

knz committed:
kvclient,kvserver,execinfrapb,server,log: use constructor for initial AmbientContext

Release note: None

knz committed:
testutils: delegate MakeAmbientCtx to an ambientcontext constructor

Release note: None

knz committed:
server,serverutils: add an `AmbientCtx()` accessor on `TestTenantInterface`

Release note: None

knz committed:
sql,kvtenantccl: use existing AmbientContexts where applicable in tests

Release note: None

knz committed:
*: use `testutils.MakeAmbientCtx()` where appropriate

so that we avoid manual construction of AmbientContext.

Release note: None

knz committed:
kvcoord: properly reuse existing AmbientContexts in tests

Release note: None

knz committed:
kvserver: reuse AmbientContexts where appropriate in tests

Release note: None

knz committed:
server,testcluster: use client AmbientContexts in tests where appropriate

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72912)
<!-- Reviewable:end -->

tbg reviewed (approved):
I didn't scrutinize the code line-by-line, but note that you added a lint which is more useful than any close read I could provide anyway.

Sorry to have let this sit.

knz committed:
kvserver: avoid more naked AmbientContexts

Release note: None

andreimatei commented on pkg/testutils/lint/lint_test.go:
```diff
@@ -546,6 +546,38 @@ func TestLint(t *testing.T) {
+		}
+
+		if err := stream.ForEach(filter, func(s string) {
+			t.Errorf("\n%s <- forbidden; reuse an existing AC when possible, or use one of the constructor functions instead: log.MakeServerAC(), log.MakeClientAC() or testutils.MakeAC()", s)
```
please make sure there's a pragma that can be used to inhibit the linter at a call site.

andreimatei replied:
please add some comments to this linter, and the commit message, say why you're doing this

andreimatei replied:
please make it more clear what's being prohibited - in a comment on the linter, and in the error message. I don't think it's obvious that constructing the struct is what's being rejected.

andreimatei reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72912)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @RaduBerinde)

___
*[pkg/acceptance/localcluster/cluster.go, line 276 at r18](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3_r4XAso28_y3G1ib:-Mr3_r4XAso28_y3G1ic:b3qh1t0) ([raw file](https://github.com/cockroachdb/cockroach/blob/e2f35aec47ddece53a6e09d74d468e58a3984ddf/pkg/acceptance/localcluster/cluster.go#L276)):*
> ```Go
> 	rpcCtx := rpc.NewContext(rpc.ContextOptions{
> 		TenantID:   roachpb.SystemTenantID,
> 		AmbientCtx: testutils.MakeAmbientCtx(),
> ```

I think this `testutils.MakeAmbientCtx()` is bad news because it creates a `Tracer` internally. This was always fishy, but it became more brittle recently because I've made it illegal to mix spans created by different tracers within a single trace (I've had my reasons for doing this now that I'm trying to make tracing real; it was always pretty broken). And it's very easy to end up mixing tracers if you have more than one. So we should fight against creating tracers; as much as possible, there should be one tracer passed in everywhere.
___
*[pkg/cli/start.go, line 766 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3XgYhEo4G2hQvsEtC:-Mr3XgYhEo4G2hQvsEtD:biyu82z) ([raw file](https://github.com/cockroachdb/cockroach/blob/d1c137d279e03f7190f228e78db5c1a74b93df17/pkg/cli/start.go#L766)):*
> ```Go
> 			// Don't use shutdownCtx because this is in a goroutine that may
> 			// still be running after shutdownCtx's span has been finished.
> 			drainCtx := ambientCtx.AnnotateCtx(context.Background())
> ```

are the other uses of `ambientCtx` in this method also broken?
___
*[pkg/server/server.go, line 2932 at r22](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3bta_3WUXZ60AI0tT:-Mr3btaa3AjhDDMAW2yr:by55hxv) ([raw file](https://github.com/cockroachdb/cockroach/blob/186f9b8b2ebddff768a8de5514a6057de6a6f4a8/pkg/server/server.go#L2932)):*
> ```Go
> 
> // AmbientCtx retrieves the ambient context for this server.
> func (s *Server) AmbientCtx() log.AmbientContext {
> ```

it seems bizarre to me to expose an AmbientCtx. The AmbientCtx is kinda by nature internal I think; it's supposed to be used to background operations originating in the inner conmponent. Where are you using all these new methods? If they're indeed needed, consider squashing this commit with some uses.
___
*[pkg/util/log/ambient_context.go, line 194 at r16](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3YMYk6J8D47W-10da:-Mr3YMYk6J8D47W-10db:b27ksa0) ([raw file](https://github.com/cockroachdb/cockroach/blob/b155b24644376528d48ecfdac3f139625d0485ed/pkg/util/log/ambient_context.go#L194)):*
> ```Go
> // MakeClientAmbientContext creates an AmbientContext for use by
> // client commands.
> func MakeClientAmbientContext(tracer *tracing.Tracer) AmbientContext {
> ```

I'm not sure the name `Client` here deserves to exist. If it does exist, let's explain more what a "client command" is. But I think it'd be better if this was just named `MakeAmbientContext`. The "client commands" seem to use this only to pass to `NewRPCContext`, and I think that an `RPCContext` shouldn't even necessarily need an `AmbientCtx` (as it does now). I think it should only have a `Tracer`. AFAICS, the `AmbientCtx` is used only for this ctx here:
https://github.com/cockroachdb/cockroach/blob/775c2b0b0302ebf4b0c8c8515e015b793b8585ce/pkg/rpc/context.go#L449
I think for that one, it should just use the `ctx` passed to `NewContext`.
___
*[pkg/util/log/ambient_context.go, line 199 at r17](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3_L8xBjaduUGJkngQ:-Mr3_L8xBjaduUGJkngR:bakocmd) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd895286c26fca26f20d65ccf3020a09e3f67e32/pkg/util/log/ambient_context.go#L199)):*
> ```Go
> 
> // MakeDummyAmbientContext creates an AmbientContext for use in tests.
> func MakeDummyAmbientContext(tracer *tracing.Tracer) AmbientContext {
> ```

What's "dummy" about it? I think this should be the same as `MakeClientAmbientContext` - simply `MakeClientAmbientContext`.


<!-- Sent from Reviewable.io -->

knz commented:
Before I engage with the review comments, I'd like to make it clear that **I'm only doing this work in response to a nudge by @tbg  in [this comment](https://github.com/cockroachdb/cockroach/pull/72702#issuecomment-970200335)**.

I don't have particular stakes in this game, so if someone doesn't like what's happening in this PR, I will only work further if there's a concrete, actionable suggestion of what needs to happen instead. (i.e. I will not _think_ about what's best and instead close the PR if it's unclear what I should be doing)

Thanks in advance.

knz committed:
*: rename `MakeDummyAmbientContext` to `MakeTestingAmbientContext`

Release note: None

knz committed:
rpc,*: mandate a context.Context arg to rpc.NewContext

Recommended/suggested by @andreimatei.
Makes it possible for rpc.Context to ignore the existence
of `log.AmbientContext`.

Release note: None

knz reviewed (commented):
Ok I have been able to do a few useful things in response to Andrei's comments. Added some fixup commits at the end to avoid a humongous rebase. RFAL.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-:-Mr7NXn357qGkAn_54Au:b-o0657e)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @gh-casper, and @RaduBerinde)

___
*[pkg/acceptance/localcluster/cluster.go, line 276 at r18](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3_r4XAso28_y3G1ib:-Mr73-PN-MWD_n-0DlKI:btzhrko) ([raw file](https://github.com/cockroachdb/cockroach/blob/e2f35aec47ddece53a6e09d74d468e58a3984ddf/pkg/acceptance/localcluster/cluster.go#L276)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I think this `testutils.MakeAmbientCtx()` is bad news because it creates a `Tracer` internally. This was always fishy, but it became more brittle recently because I've made it illegal to mix spans created by different tracers within a single trace (I've had my reasons for doing this now that I'm trying to make tracing real; it was always pretty broken). And it's very easy to end up mixing tracers if you have more than one. So we should fight against creating tracers; as much as possible, there should be one tracer passed in everywhere.
</blockquote></details>

Yes, I agree! And I am willing to help with that.
However, if at all possible I'd like to avoid doing so in this PR. The code as I'm leaving here is is no way worse than the original. We can take another pass in a later PR to replace all the calls to `testutils.MakeAmbientCtx()` by something better.
___
*[pkg/cli/start.go, line 766 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3XgYhEo4G2hQvsEtC:-Mr73Ctd2ixNO4MewKlU:boi2kso) ([raw file](https://github.com/cockroachdb/cockroach/blob/d1c137d279e03f7190f228e78db5c1a74b93df17/pkg/cli/start.go#L766)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

are the other uses of `ambientCtx` in this method also broken?
</blockquote></details>

I don't understand. The only one is at the beginning, before the Server instance is instantiated. Obviously at that point we don't have a server yet so we can't use s.AnnotateCtx.

We want s.AnnotateCtx here because it may have more details about the server than the ambientCtx at the beginning.
___
*[pkg/server/server.go, line 2932 at r22](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3bta_3WUXZ60AI0tT:-Mr7AsCYEe_oDmtP4Kip:b-l8vm5p) ([raw file](https://github.com/cockroachdb/cockroach/blob/186f9b8b2ebddff768a8de5514a6057de6a6f4a8/pkg/server/server.go#L2932)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

it seems bizarre to me to expose an AmbientCtx. The AmbientCtx is kinda by nature internal I think; it's supposed to be used to background operations originating in the inner conmponent. Where are you using all these new methods? If they're indeed needed, consider squashing this commit with some uses.
</blockquote></details>

This is only used used in tests, every time a test instantiates a component that needs the same AmbientContext.

It's much more readable to write `s.AmbientCtx()` than `s.(*server.TestServer).Cfg.AmbientCtx`.

I added a commit to rename this to `TestingAmbientCtx` to highlight the purpose.
___
*[pkg/util/log/ambient_context.go, line 194 at r16](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3YMYk6J8D47W-10da:-Mr7LRjLDYV-uR2nT79W:b-kbd6dv) ([raw file](https://github.com/cockroachdb/cockroach/blob/b155b24644376528d48ecfdac3f139625d0485ed/pkg/util/log/ambient_context.go#L194)):*
So I think we need to have a separate constructor. The primary reason is that this sentence is false:

>  I think that an `rpc.Context` shouldn't even necessarily need an AmbientCtx (as it does now)

It does - the `rpc.Context` used by *servers* needs to annotate its contexts with the server identifiers and other background states. So we need an AmbientContext there in any case.

What is sad (but can't be fixed in this PR) is that we use the same `rpc.Context` type for both servers and clients. It's sad because clients don't need all the complexity we've built into `rpc.Context`, and would benefit from a much simpler type. We want to fix that, but then again not in this PR. Let's file an issue for this: #73967

However, I am sympathetic to the need to clarify this in `ambient_context.go`. I added some explanatory comment for this with a reference to the new issue. 
___
*[pkg/util/log/ambient_context.go, line 199 at r17](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3_L8xBjaduUGJkngQ:-Mr7DmTQBBFgRZe2RXjR:b1n9p0a) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd895286c26fca26f20d65ccf3020a09e3f67e32/pkg/util/log/ambient_context.go#L199)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

What's "dummy" about it? I think this should be the same as `MakeClientAmbientContext` - simply `MakeClientAmbientContext`.
</blockquote></details>

I agree the word "dummy" was mistaken. I meant to write "Testing" here - this is a constructor for use in tests. Added a commit to fix that. Good call.
___
*[pkg/testutils/lint/lint_test.go, line 569 at r29](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mon_wkbGAM6R9nmewTl-r29-569:-Mr74QQGCvlF5kfWKPln:b-10c1vd) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7eb04443f922c4d525790c62dc27938d2d4ceb7/pkg/testutils/lint/lint_test.go#L569)):*
> please make sure there's a pragma that can be used to inhibit the linter at a call site.

I'm not sure that's necessary here. One can always use a constructor and modify the AC afterwards.

> please add some comments to this linter, and the commit message, say why you're doing this
> please make it more clear what's being prohibited - in a comment on the linter, and in the error message. I don't think it's obvious that constructing the struct is what's being rejected.

Good idea. Also added a reference to the tech note.

Also I added an additional commit, which *replaces the linter altogether* by a change to the type definition to make it non-constructible. I think it's even better this way.


<!-- Sent from Reviewable.io -->

knz commented:
Is this good to go now?

tbg commented:
I don't want to preempt Andrei's next review pass, but I am in favor of merging this PR. There don't seem to be fundamental disagreements on the approach (I like the non-constructability, good call) and a follow-up PR can address any further review comments.

andreimatei reviewed (commented):
> I don't have particular stakes in this game, so if someone doesn't like what's happening in this PR, I will only work further if there's a concrete, actionable suggestion of what needs to happen instead. (i.e. I will not think about what's best and instead close the PR if it's unclear what I should be doing)

I don't want to particularly discourage you from doing things around here, but, personally, I'm not convinced this is all worth it. I think that generally this PR came to be from work to get more people to use more log tags - which I think is usually noble - but I don't exactly see how the various changes in this PR help that goal.

> Added some fixup commits at the end to avoid a humongous rebase.

There's at least three things that seem to go back and forth in this PR:
1) the linter about not construction the `AmbientCtx` is first introduced, then deleted
2) one of the ctors is first called  `MakeDummyAmbientContext`, and then renamed to `MakeTestingAmbientContext`
3) the changes in `adminTest` first use the wrong ctor, and then switch to a different one.
 Let's avoid such churn... You chose to send out a PR with a lot of commits, I think you need to deal with the rebases...

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-:-MrNgAKA5vvq9D3DK7b5:bb11wzg)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gh-casper, @knz, and @RaduBerinde)

___
*[pkg/acceptance/localcluster/cluster.go, line 276 at r18](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3_r4XAso28_y3G1ib:-MrNhLd74dA8hJZ8qI2t:b32wtd) ([raw file](https://github.com/cockroachdb/cockroach/blob/e2f35aec47ddece53a6e09d74d468e58a3984ddf/pkg/acceptance/localcluster/cluster.go#L276)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Yes, I agree! And I am willing to help with that.
However, if at all possible I'd like to avoid doing so in this PR. The code as I'm leaving here is is no way worse than the original. We can take another pass in a later PR to replace all the calls to `testutils.MakeAmbientCtx()` by something better.
</blockquote></details>

ok
___
*[pkg/cli/start.go, line 766 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3XgYhEo4G2hQvsEtC:-MrN_KcI-RELKy_yR70A:b6e4sgf) ([raw file](https://github.com/cockroachdb/cockroach/blob/d1c137d279e03f7190f228e78db5c1a74b93df17/pkg/cli/start.go#L766)):*
> The only one is at the beginning, before the Server instance is instantiated.

There's also this one:
https://github.com/cockroachdb/cockroach/blob/0e6abb5778079f9560eae51b33979654c0894639/pkg/cli/start.go#L702
___
*[pkg/server/server.go, line 2932 at r22](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3bta_3WUXZ60AI0tT:-MrNhV9e1wOHsvsr6eqL:b-12iuyj) ([raw file](https://github.com/cockroachdb/cockroach/blob/186f9b8b2ebddff768a8de5514a6057de6a6f4a8/pkg/server/server.go#L2932)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

This is only used used in tests, every time a test instantiates a component that needs the same AmbientContext.

It's much more readable to write `s.AmbientCtx()` than `s.(*server.TestServer).Cfg.AmbientCtx`.

I added a commit to rename this to `TestingAmbientCtx` to highlight the purpose.


</blockquote></details>

Then let's put the method on `TestServer` (and drop the `Testing` from the method name).
___
*[pkg/util/log/ambient_context.go, line 194 at r16](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3YMYk6J8D47W-10da:-MrNaPhE4_QP4g1GOxrJ:b-pkc1r0) ([raw file](https://github.com/cockroachdb/cockroach/blob/b155b24644376528d48ecfdac3f139625d0485ed/pkg/util/log/ambient_context.go#L194)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

So I think we need to have a separate constructor. The primary reason is that this sentence is false:

>  I think that an `rpc.Context` shouldn't even necessarily need an AmbientCtx (as it does now)

It does - the `rpc.Context` used by *servers* needs to annotate its contexts with the server identifiers and other background states. So we need an AmbientContext there in any case.

What is sad (but can't be fixed in this PR) is that we use the same `rpc.Context` type for both servers and clients. It's sad because clients don't need all the complexity we've built into `rpc.Context`, and would benefit from a much simpler type. We want to fix that, but then again not in this PR. Let's file an issue for this: #73967

However, I am sympathetic to the need to clarify this in `ambient_context.go`. I added some explanatory comment for this with a reference to the new issue. 

</blockquote></details>

I still think that `rpc.Context` can easily avoid the need for an `AmbientCtx`, be it used by servers or by clients. I think the `AmbientCtx` is only used here:
https://github.com/cockroachdb/cockroach/blob/775c2b0b0302ebf4b0c8c8515e015b793b8585ce/pkg/rpc/context.go#L449

That's in the constructor; instead of doing `AmbientCtx.AnnotateCtx(context.Background())`, I think the ctor (`NewContext()`) should take a `ctx`.

The `rpc.Context` needs a `Tracer` in a couple of other places - so I think we should put a `Tracer` in `ContextOptions`, instead of the `AmbientCtx`. I think we shouldn't proliferate the `AmbientCtx` to places that don't truly get a benefit from it - particularly to this bloated `rpc.Context` library.

---

If you don't want to do this changes, I'd still like to avoid the `Client` in `MakeClientAmbientContext`. Let's just call this constructor `MakeAmbientContext`, and try to keep this file as agnostic about its uses as possible.
___
*[pkg/util/log/ambient_context.go, line 199 at r17](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3_L8xBjaduUGJkngQ:-MrNhihYDbt5p58L0e3Z:bx8oewn) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd895286c26fca26f20d65ccf3020a09e3f67e32/pkg/util/log/ambient_context.go#L199)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I agree the word "dummy" was mistaken. I meant to write "Testing" here - this is a constructor for use in tests. Added a commit to fix that. Good call.
</blockquote></details>

This is exactly like `MakeClientAmbientContext` above... Let's just have a `MakeAmbientContext()`. This library doesn't need to be aware of every single caller.
___
*[pkg/util/log/ambient_context.go, line 211 at r34](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrNiLYFBNnt8hteneJA:-MrNiLYFBNnt8hteneJB:bpj1nwg) ([raw file](https://github.com/cockroachdb/cockroach/blob/b49a258b8ae9ee8430bebeeec8983ab2da107704/pkg/util/log/ambient_context.go#L211)):*
> ```Go
> // See: https://github.com/cockroachdb/cockroach/issues/73967
> func MakeClientAmbientContext(tracer *tracing.Tracer) AmbientContext {
> 	// TODO(knz): add some annotations in the AmbientContext that
> ```

I would probably be against that. The logs for a client command don't need to highlight in every line that this is a client command.
___
*[pkg/testutils/lint/lint_test.go, line 569 at r29](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mon_wkbGAM6R9nmewTl-r29-569:-MrNkg1r-E2fbw5jwAsr:b-2a5x86) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7eb04443f922c4d525790c62dc27938d2d4ceb7/pkg/testutils/lint/lint_test.go#L569)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

> please make sure there's a pragma that can be used to inhibit the linter at a call site.

I'm not sure that's necessary here. One can always use a constructor and modify the AC afterwards.

> please add some comments to this linter, and the commit message, say why you're doing this
> please make it more clear what's being prohibited - in a comment on the linter, and in the error message. I don't think it's obvious that constructing the struct is what's being rejected.

Good idea. Also added a reference to the tech note.

Also I added an additional commit, which *replaces the linter altogether* by a change to the type definition to make it non-constructible. I think it's even better this way.

</blockquote></details>

Let's not go overboard with this stuff. There's many types where the zero value is fairly unusable, and that have required fields. We don't write linters for them or play tricks to make them hard to instantiate (which you can't even do in go btw; one can always create a zero value). And in particular, with the `AmbientCtx`, the zero value *is* sometimes technically usable. I personally don't feel the need for either the linter, and definitely not for adding extra code for the `privateAmbientCtx`. In particular, that changes required making `Tracer()` a method instead of a field, which is really a lot of code churn for no reason. Let's keep `AmbientCtx` dead simple.


<!-- Sent from Reviewable.io -->

knz committed:
cli/start: clarify AmbientCtx usage in comment

Release note: None

knz reviewed (commented):
> the linter about not construction the AmbientCtx is first introduced, then deleted

removed both now

> Let's avoid such churn... You chose to send out a PR with a lot of commits, I think you need to deal with the rebases...

I performed the rebases, to combine commits that belong together.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-:-MrU-Dt-2xKW4tnag1lg:bc64fz8)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @gh-casper, and @RaduBerinde)

___
*[pkg/cli/start.go, line 766 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3XgYhEo4G2hQvsEtC:-MrTC0sy0hHgOy5ejMik:b6iuyqv) ([raw file](https://github.com/cockroachdb/cockroach/blob/d1c137d279e03f7190f228e78db5c1a74b93df17/pkg/cli/start.go#L766)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

> The only one is at the beginning, before the Server instance is instantiated.

There's also this one:
https://github.com/cockroachdb/cockroach/blob/0e6abb5778079f9560eae51b33979654c0894639/pkg/cli/start.go#L702

</blockquote></details>

Yeah unfortunately the `*Server` may not be available yet at that point. Clarified in comment.
___
*[pkg/server/server.go, line 2932 at r22](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3bta_3WUXZ60AI0tT:-MrTC6AI85A--H4b3Nj4:b-k7v037) ([raw file](https://github.com/cockroachdb/cockroach/blob/186f9b8b2ebddff768a8de5514a6057de6a6f4a8/pkg/server/server.go#L2932)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

Then let's put the method on `TestServer` (and drop the `Testing` from the method name).
</blockquote></details>

Ah, good point. Done.
___
*[pkg/util/log/ambient_context.go, line 194 at r16](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3YMYk6J8D47W-10da:-MrTtvhc6F0JyPNY0Ekm:bn0i2qo) ([raw file](https://github.com/cockroachdb/cockroach/blob/b155b24644376528d48ecfdac3f139625d0485ed/pkg/util/log/ambient_context.go#L194)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I still think that `rpc.Context` can easily avoid the need for an `AmbientCtx`, be it used by servers or by clients. I think the `AmbientCtx` is only used here:
https://github.com/cockroachdb/cockroach/blob/775c2b0b0302ebf4b0c8c8515e015b793b8585ce/pkg/rpc/context.go#L449

That's in the constructor; instead of doing `AmbientCtx.AnnotateCtx(context.Background())`, I think the ctor (`NewContext()`) should take a `ctx`.

The `rpc.Context` needs a `Tracer` in a couple of other places - so I think we should put a `Tracer` in `ContextOptions`, instead of the `AmbientCtx`. I think we shouldn't proliferate the `AmbientCtx` to places that don't truly get a benefit from it - particularly to this bloated `rpc.Context` library.

---

If you don't want to do this changes, I'd still like to avoid the `Client` in `MakeClientAmbientContext`. Let's just call this constructor `MakeAmbientContext`, and try to keep this file as agnostic about its uses as possible.
</blockquote></details>

Ok, I was convinced. I changed the `rpc.NewContext` to take a ctx argument and simplified everything accordingly. Added a new commit to do this.
___
*[pkg/util/log/ambient_context.go, line 199 at r17](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mr3_L8xBjaduUGJkngQ:-MrTxSf1EVz1K4MboP5S:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd895286c26fca26f20d65ccf3020a09e3f67e32/pkg/util/log/ambient_context.go#L199)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

This is exactly like `MakeClientAmbientContext` above... Let's just have a `MakeAmbientContext()`. This library doesn't need to be aware of every single caller.
</blockquote></details>

Done.
___
*[pkg/util/log/ambient_context.go, line 211 at r34](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrNiLYFBNnt8hteneJA:-MrTxQWJCHeSfEBpTdml:by63xp1) ([raw file](https://github.com/cockroachdb/cockroach/blob/b49a258b8ae9ee8430bebeeec8983ab2da107704/pkg/util/log/ambient_context.go#L211)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I would probably be against that. The logs for a client command don't need to highlight in every line that this is a client command.
</blockquote></details>

Gotcha. Removed.
___
*[pkg/testutils/lint/lint_test.go, line 569 at r29](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mon_wkbGAM6R9nmewTl-r29-569:-MrTxUrc2ByaNfxkX68n:b-e5sa3p) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7eb04443f922c4d525790c62dc27938d2d4ceb7/pkg/testutils/lint/lint_test.go#L569)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

Let's not go overboard with this stuff. There's many types where the zero value is fairly unusable, and that have required fields. We don't write linters for them or play tricks to make them hard to instantiate (which you can't even do in go btw; one can always create a zero value). And in particular, with the `AmbientCtx`, the zero value *is* sometimes technically usable. I personally don't feel the need for either the linter, and definitely not for adding extra code for the `privateAmbientCtx`. In particular, that changes required making `Tracer()` a method instead of a field, which is really a lot of code churn for no reason. Let's keep `AmbientCtx` dead simple.
</blockquote></details>

We discussed this offline. No need to remove the ability to construct. 
So as agreed I removed the offending two commits (both the linter and the one introducing the private field).


<!-- Sent from Reviewable.io -->

andreimatei reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> but please go through my comments

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-:-MrcIt2_-SgM2zIXGviZ:bnd4jbx)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei, @gh-casper, @knz, @RaduBerinde, and @stevendanna)

___
*[-- commits, line 13 at r37](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc-LaoEOTSiwo5GBV2):*
consider dropping this commit now that you have the `rpc,*: mandate a context.Context arg to rpc.NewContext` one.
___
*[-- commits, line 32 at r40](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc1ABw9g1BMuYXw6mU):*
consider dropping this commit now that you have the `rpc,*: mandate a context.Context arg to rpc.NewContext` one.
___
*[-- commits, line 57 at r45](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc6bhJDsZ11sAti9D8):*
I won't make a big deal out of this, but I'm not in favor of most of this commit. You're replacing a bunch of `log.AmbientContext{Tracer: tracing.NewTracer()}` with `testutils.MakeAmbientCtx()`. As discussed before, `testutils.MakeAmbientCtx()` is generally a bad idea because it creates a `Tracer` under the hood, and if this `Tracer` ends up actually being used, chances are it's going to crash because it'll end up mixing with some other `Tracer`. The code that you're modifying has the same problem - so either all these tracers don't actually end up being used (in which case it'd be better if we used the zero value `AmbientContext{}`), or it does end up being used but, by happenstance, the trace in question doesn't end up having more than one span.
On some other commit you've done a similar replacement on the argument that "it's not making the code any worse" than it already was. On a low scale, I don't care enough to argue, but on a large scale (i.e. this commit), the way I see it, these replacements are at best a lot of churn for no reason (at least I don't see the reason) or, at worse, a lot of churn that makes the code a little worse by slightly hiding the fact that a probably-unintended Tracer is being built.
___
*[pkg/rpc/context.go, line 378 at r50](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrcHzJC13LK8fmD96qa:-MrcHzJD8qmOGJS2miH-:bb7r7eq) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c1c33341a98724f89725799036e0b835c4c56c9/pkg/rpc/context.go#L378)):*
> ```Go
> type ContextOptions struct {
> 	TenantID   roachpb.TenantID
> 	AmbientCtx log.AmbientContext
> ```

woot
___
*[pkg/sql/pgwire/fuzz.go, line 32 at r45](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc5ijp8YLMGRD985tH:-Mrc5ijp8YLMGRD985tI:b-nz25lg) ([raw file](https://github.com/cockroachdb/cockroach/blob/c608fed4276b9636292d79cd1e766eb374ca108e/pkg/sql/pgwire/fuzz.go#L32)):*
> ```Go
> func FuzzServeConn(data []byte) int {
> 	s := MakeServer(
> 		testutils.MakeAmbientCtx(),
> ```

I think the zero value is better. As discussed elsewhere, `testutils.MakeAmbientCtx()` is generally bad because it creates a `Tracer`, and generally you want the `Tracer` passed in. You've introduced more uses of `MakeAmbientCtx()` instead of `AmbientContext{Tracer: NewTracer()}` on the argument that you're not making the code any worse than it was, but in this case you are making it worse in my opinion, as you're introducing a `Tracer` where there was none. When the zero value is sufficient, I think there's no reason not to use it.
___
*[pkg/testutils/localtestcluster/local_test_cluster.go, line 54 at r46](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc917V9M8HPmghhUfd:-Mrc917W0XonzfeIBv7L:b-3enbon) ([raw file](https://github.com/cockroachdb/cockroach/blob/8852ed1692ad26cf65e75632e422e87e4bba28d3/pkg/testutils/localtestcluster/local_test_cluster.go#L54)):*
> ```Go
> // in that although it uses a distributed sender, there is no RPC traffic.
> type LocalTestCluster struct {
> 	*log.AmbientContext
> ```

I personally would vote to not embed, and give the field a name instead. Embedding makes its harder to find uses of the thing, and I've found myself needing to find the uses of `AmbientContexts` - particularly to find people grabbing the Tracer.

Also, how come this is a pointer? That's unusual, instead it? If there's indeed a good reason for that, consider giving it a comment.

Also, I'm confused about why you've done this change. How come the `LocalTestCluster`? What's supposed to be in this `AmbientCtx`? Within a minute of complaining above that uses of embedded fields are not easy to find, I wanted to see who uses this, and so I've commented the field out and recompiled all the tests (took 30% of my battery lol). I see that it's only used in `LocalTestCLuster.Start()`, and so I'm not sure why this needs to be a field on the struct. If it doesn't need to be a field, I think we should avoid it.
___
*[pkg/util/log/ambient_context.go, line 198 at r38](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc030j8Jp4ifwk405S:-Mrc030j8Jp4ifwk405T:b-lvqzhn) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f44ab38c1fc17ee58d334a3545e38ef96a2f00d/pkg/util/log/ambient_context.go#L198)):*
> ```Go
> }
> 
> // MakeDummyAmbientContext creates an AmbientContext for use in tests.
> ```

consider squashing this commit with the `rename `MakeDummyAmbientContext` to `MakeTestingAmbientContext` one.

I would still like this to be called `MakeAmbientContext`, as there's nothing particularly `Testing` about it (particularly not from the perspective of this `log` library`, but I'll let it go. 
___
*[pkg/util/log/ambient_context.go, line 208 at r41](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc2SCs0GS8VN1X-gjZ:-Mrc2SCs0GS8VN1X-gj_:b-u8z427) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f5bde1fdc9fde41da1e47432d322b65e589dcd/pkg/util/log/ambient_context.go#L208)):*
> ```Go
> 	tracer *tracing.Tracer, idProvider ServerIdentificationPayload,
> ) AmbientContext {
> 	// TODO(knz): add the server identifier containers here as mandatory
> ```

for my curiosity, what are the "server identifier containers, if not the `ServerIdentificationPayload`? Consider adding more words to the comment hinting at why we'd want more than the `ServerIdentificationPayload`.
___
*[pkg/kv/kvclient/kvcoord/dist_sender_test.go, line 4578 at r46](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrcEU2t3pbXKo1plskA:-MrcEU2u2vOADaKUKrfL:b-is7onp) ([raw file](https://github.com/cockroachdb/cockroach/blob/8852ed1692ad26cf65e75632e422e87e4bba28d3/pkg/kv/kvclient/kvcoord/dist_sender_test.go#L4578)):*
> ```Go
> 		t.Run(tc.name, func(t *testing.T) {
> 			st := cluster.MakeTestingClusterSettings()
> 			ambientCtx := testutils.MakeAmbientCtx()
> ```

I think this is backwards. We need to create the `Tracer` early, but we don't need to create the `AmbientCtx` early. I think the `NewTracer()` call was good here, and I think we should use `MakeTestingAmbientCtx(tr)` below, when creating the `DistSenderConfig`.
___
*[pkg/kv/kvserver/allocator_test.go, line 7315 at r47](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrcFC0A9lGQr_92zvZg:-MrcFC0A9lGQr_92zvZh:botc5xl) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa11872502d9d8bc5b6c65792cb91312a14b3dba/pkg/kv/kvserver/allocator_test.go#L7315)):*
> ```Go
> 
> 	st := cluster.MakeTestingClusterSettings()
> 	ambientCtx := testutils.MakeAmbientCtx()
> ```

nit: I personally would prefer it if we left `tr := NewTracer()` here, and used `MakeTestingAmbientContext(tr)` twice below, once for `rpcContext` and once for `storePool`. The creation of those two structs doesn't need to share an `AmbientCtx` (particularly not an empty one), so I think it's better if they both create their own (so that the code doesn't suggest there's some coupling between them).
___
*[pkg/kv/kvserver/client_raft_test.go, line 3037 at r43](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc3FO29pRZoH18Vmgl:-Mrc3FO29pRZoH18Vmgm:b7qc0r9) ([raw file](https://github.com/cockroachdb/cockroach/blob/8745ce89debe67dffe16f9a34137914b5d037eef/pkg/kv/kvserver/client_raft_test.go#L3037)):*
> ```Go
> 	// there are two (this one and the one actually used by the store).
> 	fromTransport := kvserver.NewRaftTransport(
> 		tc.Servers[0].AmbientCtx(),
> ```

I'm not sure this is an improvement. I'm not immediately sure what this test does, but this raft transport runs kinda outside of the first node. Consider giving it a "test" log tag - either in addition to, or instead of, the node's tags. 
___
*[pkg/testutils/testcluster/testcluster_test.go, line 210 at r48](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrcGD33BPE8me2Ng7sx:-MrcGD347u7Jsk2x_5-P:b-fxkzpt) ([raw file](https://github.com/cockroachdb/cockroach/blob/e7ce5a43326702475819662a54da785c3405b241/pkg/testutils/testcluster/testcluster_test.go#L210)):*
> ```Go
> 	}
> 
> 	tr := tc.Server(0).TracerI().(*tracing.Tracer)
> ```

is this `tr` used? I think this commit won't compile by itself.


<!-- Sent from Reviewable.io -->

knz committed:
*: remove `testutils.MakeAmbientCtx`

This was recommended by @andreimatei. The name was hiding too much
the fact it's creating a tracer under the hood.
Removed in favor of a new `log.MakeTestingAmbientCtxWithNewTracer`,
with an explanatory comment that indicates how this really should
not be used.

Release note: None

knz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72912)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei, @gh-casper, @RaduBerinde, and @stevendanna)

___
*[-- commits, line 13 at r37](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc-LaoEOTSiwo5GBV2):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

consider dropping this commit now that you have the `rpc,*: mandate a context.Context arg to rpc.NewContext` one.
</blockquote></details>

Done.
___
*[-- commits, line 32 at r40](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc1ABw9g1BMuYXw6mU):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

consider dropping this commit now that you have the `rpc,*: mandate a context.Context arg to rpc.NewContext` one.
</blockquote></details>

Done.
___
*[-- commits, line 57 at r45](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc6bhJDsZ11sAti9D8):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I won't make a big deal out of this, but I'm not in favor of most of this commit. You're replacing a bunch of `log.AmbientContext{Tracer: tracing.NewTracer()}` with `testutils.MakeAmbientCtx()`. As discussed before, `testutils.MakeAmbientCtx()` is generally a bad idea because it creates a `Tracer` under the hood, and if this `Tracer` ends up actually being used, chances are it's going to crash because it'll end up mixing with some other `Tracer`. The code that you're modifying has the same problem - so either all these tracers don't actually end up being used (in which case it'd be better if we used the zero value `AmbientContext{}`), or it does end up being used but, by happenstance, the trace in question doesn't end up having more than one span.
On some other commit you've done a similar replacement on the argument that "it's not making the code any worse" than it already was. On a low scale, I don't care enough to argue, but on a large scale (i.e. this commit), the way I see it, these replacements are at best a lot of churn for no reason (at least I don't see the reason) or, at worse, a lot of churn that makes the code a little worse by slightly hiding the fact that a probably-unintended Tracer is being built.
</blockquote></details>

Understood. I agree on principle, but just cancelling the commit altogether is a bit too much work - to do it properly, I might as well go through all the tests that use the API and figure out which tracer they'd better used.

So instead, I added a new commit which removes `testutils.MakeAmbientCtx` altogether with a comment about what needs to happen in the future.
___
*[pkg/sql/pgwire/fuzz.go, line 32 at r45](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc5ijp8YLMGRD985tH:-MrwevegBFTM1hi-Imcw:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c608fed4276b9636292d79cd1e766eb374ca108e/pkg/sql/pgwire/fuzz.go#L32)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I think the zero value is better. As discussed elsewhere, `testutils.MakeAmbientCtx()` is generally bad because it creates a `Tracer`, and generally you want the `Tracer` passed in. You've introduced more uses of `MakeAmbientCtx()` instead of `AmbientContext{Tracer: NewTracer()}` on the argument that you're not making the code any worse than it was, but in this case you are making it worse in my opinion, as you're introducing a `Tracer` where there was none. When the zero value is sufficient, I think there's no reason not to use it.
</blockquote></details>

Done.
___
*[pkg/testutils/localtestcluster/local_test_cluster.go, line 54 at r46](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc917V9M8HPmghhUfd:-Mrwg8X4B3L4Tq2sZQmI:bmo5q0m) ([raw file](https://github.com/cockroachdb/cockroach/blob/8852ed1692ad26cf65e75632e422e87e4bba28d3/pkg/testutils/localtestcluster/local_test_cluster.go#L54)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I personally would vote to not embed, and give the field a name instead. Embedding makes its harder to find uses of the thing, and I've found myself needing to find the uses of `AmbientContexts` - particularly to find people grabbing the Tracer.

Also, how come this is a pointer? That's unusual, instead it? If there's indeed a good reason for that, consider giving it a comment.

Also, I'm confused about why you've done this change. How come the `LocalTestCluster`? What's supposed to be in this `AmbientCtx`? Within a minute of complaining above that uses of embedded fields are not easy to find, I wanted to see who uses this, and so I've commented the field out and recompiled all the tests (took 30% of my battery lol). I see that it's only used in `LocalTestCLuster.Start()`, and so I'm not sure why this needs to be a field on the struct. If it doesn't need to be a field, I think we should avoid it.
</blockquote></details>

It's used in txn_coord_sender_test.go.
But point taken, I moved it to a named field and removed the reference.
___
*[pkg/util/log/ambient_context.go, line 198 at r38](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc030j8Jp4ifwk405S:-Mrwkd4b7CvGP40WAXFV:bv35cuj) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f44ab38c1fc17ee58d334a3545e38ef96a2f00d/pkg/util/log/ambient_context.go#L198)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

consider squashing this commit with the `rename `MakeDummyAmbientContext` to `MakeTestingAmbientContext` one.

I would still like this to be called `MakeAmbientContext`, as there's nothing particularly `Testing` about it (particularly not from the perspective of this `log` library`, but I'll let it go. 
</blockquote></details>

I tried to squash them, but there's all the intermediate changes to introduce the call that don't play nice with the rebase.
___
*[pkg/util/log/ambient_context.go, line 208 at r41](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc2SCs0GS8VN1X-gjZ:-MrwsLK9BoApZ91L2QqL:biu0w6a) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f5bde1fdc9fde41da1e47432d322b65e589dcd/pkg/util/log/ambient_context.go#L208)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

for my curiosity, what are the "server identifier containers, if not the `ServerIdentificationPayload`? Consider adding more words to the comment hinting at why we'd want more than the `ServerIdentificationPayload`.
</blockquote></details>

This was a stray comment, from before a rebase that added the idProvider to the argument list. Removed.
___
*[pkg/kv/kvclient/kvcoord/dist_sender_test.go, line 4578 at r46](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrcEU2t3pbXKo1plskA:-MrwkkBt5mWOm13S8oww:bspotlo) ([raw file](https://github.com/cockroachdb/cockroach/blob/8852ed1692ad26cf65e75632e422e87e4bba28d3/pkg/kv/kvclient/kvcoord/dist_sender_test.go#L4578)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I think this is backwards. We need to create the `Tracer` early, but we don't need to create the `AmbientCtx` early. I think the `NewTracer()` call was good here, and I think we should use `MakeTestingAmbientCtx(tr)` below, when creating the `DistSenderConfig`.
</blockquote></details>

Good catch. Reverted.
___
*[pkg/kv/kvserver/allocator_test.go, line 7315 at r47](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrcFC0A9lGQr_92zvZg:-MrwoSMS4u-nqzMu-MYE:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa11872502d9d8bc5b6c65792cb91312a14b3dba/pkg/kv/kvserver/allocator_test.go#L7315)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

nit: I personally would prefer it if we left `tr := NewTracer()` here, and used `MakeTestingAmbientContext(tr)` twice below, once for `rpcContext` and once for `storePool`. The creation of those two structs doesn't need to share an `AmbientCtx` (particularly not an empty one), so I think it's better if they both create their own (so that the code doesn't suggest there's some coupling between them).
</blockquote></details>

Done.
___
*[pkg/kv/kvserver/client_raft_test.go, line 3037 at r43](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-Mrc3FO29pRZoH18Vmgl:-MrwlB0m3NX0wWZfgus6:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8745ce89debe67dffe16f9a34137914b5d037eef/pkg/kv/kvserver/client_raft_test.go#L3037)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I'm not sure this is an improvement. I'm not immediately sure what this test does, but this raft transport runs kinda outside of the first node. Consider giving it a "test" log tag - either in addition to, or instead of, the node's tags. 
</blockquote></details>

Done.
___
*[pkg/testutils/testcluster/testcluster_test.go, line 210 at r48](https://reviewable.io/reviews/cockroachdb/cockroach/72912#-MrcGD33BPE8me2Ng7sx:-MrwmN2PES2ijKJDHi2X:b3d8g72) ([raw file](https://github.com/cockroachdb/cockroach/blob/e7ce5a43326702475819662a54da785c3405b241/pkg/testutils/testcluster/testcluster_test.go#L210)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

is this `tr` used? I think this commit won't compile by itself.
</blockquote></details>

Oops. Good catch. Fixed.


<!-- Sent from Reviewable.io -->

knz commented:
TFYRs!

bors r=tbg,andreimatei

knz commented:
unrelated fail in logictest

bors r=tbg,andreimatei
